libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+//         {_time: 2021-06-01T03:00:00Z, _field: "temp", _value: 57},
+//     ],
+// )
+//
+// join(
+//   tables: {sf: sf_temp, ny: ny_temp},
+//   on: ["_time", "_field"]
+// )
+// ```
+```
+
+### Other Information
+
+There are other details about Flux functions that we will want to document, like the return type, the types of the parameters, and whether or not a parameter is required. We should be able to get all of that information from the function signature, and therefore shouldn't need to include it in the doc comment.
+
+### Full Example
+
+For some examples of what a complete doc comment looks like for a flux function, see the doc comments for `join()` and `aggregateWindow()` in the `universe` package.
+
+## Documenting Packages
+
+### Import path
+
+Doc comments for packages should start with the full import path for the package, wrapped in backticks. Nothing else should be included on this line.
+
+_Example:_
+
+```
+// `experimental/geo`
+```
+
+### Description
+
+Include a brief description of the package. This should read as one or more full setences.
+
+_Example:_
+
+```
+// `math`
+//
+// The Flux math package provides basic constants and mathematical functions.
+package math
+```
+
+## Documenting Flux Options
+
+### Name and Description
+
+The first line of a doc comment for a flux option should start with the name of the option wrapped in backticks, followed by a brief explanation of what the option is, or how it is used.
+
+### Extra Details
+
+Additional notes explaining how users can use the option in their code. This section should follow the same formatting rules as the `Extra Details` section for Flux functions described earlier in this document.
+
+This section may be omitted.
+
+### Code Example
+
+A markdown-formatted code block that users can copy and paste into the InfluxDB data explorer and run to see the option in action.
+
+### Full Example
+
+For an example of a complete doc comment for a Flux option, see the `enabledProfilers` option in the `profiler` package.
+
+## Documenting Builtin Constants
+
+Flux packages often come with one or more predefined constants that users can import and use in their code. Some of these constants are defined with `builtin` statements, which means their actual value is defined in Go code rather than Flux code. To document a such constant, include a line comment directly above its declaration with a string representation of the constant's value.
+
+Because this value will only ever be treated as a string in the docs, writing a description of the constant's value, rather than the literal value, is acceptable if it makes things easier to read or understand.
+
+If a constant is not defined with a builtin statement, the doc comment parser should be able to get its value from the flux code itself, and no documentation is required.
+
+_Example:_
+
+```
+// 0.693147180559945309417232121458176568075500134360255254120680009
```

Comment:
I think something like:

```
// ln2 = 0.693147180559945309417232121458176568075500134360255254120680009
builtin ln2 : float
```

I think this might be better. It matches better with the existing doc comment structure and immediately tells you what it's referencing. We can do:

```
// `ln2` = 0.693147180559945309417232121458176568075500134360255254120680009
```

If we decide to stick with backticks for the symbol name.

Reply:
I originally included something similar to this. Sorta went back and forth on it. I'll change it back.